tests/test_helpers.py

```diff
@@ -69,53 +68,56 @@ def test_get_source_from_index_without_dash():
 
 def test_parse_records():
     records = list(helpers.parse_records("tests/fixtures/sample_records.json"))
-    assert len(records) == 6
+    n_sample_records = 6
+    assert len(records) == n_sample_records
     assert isinstance(records[0], dict)
 
 
 def test_parse_deleted_records():
     records = list(
         helpers.parse_deleted_records("tests/fixtures/sample_deleted_records.txt")
     )
-    assert len(records) == 3
+    n_sample_deleted_records = 3
+    assert len(records) == n_sample_deleted_records
     assert isinstance(records[0], dict)
 
 
 def test_validate_bulk_cli_options_neither_index_nor_source_passed(
     test_opensearch_client,
 ):
-    with pytest.raises(UsageError) as error:
+    with pytest.raises(
+        UsageError, match="Must provide either an existing index name or a valid source."
+    ):
         helpers.validate_bulk_cli_options(None, None, test_opensearch_client)
-    assert "Must provide either an existing index name or a valid source." == str(
-        error.value
-    )
 
 
 def test_validate_bulk_cli_options_index_and_source_passed(test_opensearch_client):
-    with pytest.raises(UsageError) as error:
+    with pytest.raises(
+        UsageError, match="Only one of --index and --source options is allowed, not both."
+    ):
         helpers.validate_bulk_cli_options(
             "index-name", "source-name", test_opensearch_client
         )
-    assert "Only one of --index and --source options is allowed, not both." == str(
-        error.value
-    )
 
 
 @my_vcr.use_cassette("helpers/bulk_cli_nonexistent_index.yaml")
 def test_validate_bulk_cli_options_nonexistent_index_passed(test_opensearch_client):
-    with pytest.raises(BadParameter) as error:
+    with pytest.raises(
+        BadParameter, match="Index 'wrong' does not exist in the cluster."
+    ):
         helpers.validate_bulk_cli_options("wrong", None, test_opensearch_client)
-    assert "Index 'wrong' does not exist in the cluster." == str(error.value)
 
 
 @my_vcr.use_cassette("helpers/bulk_cli_no_primary_index_for_source.yaml")
 def test_validate_bulk_cli_options_no_primary_index_for_source(test_opensearch_client):
-    with pytest.raises(BadParameter) as error:
+    with pytest.raises(
+        BadParameter,
+        match=(
+            "No index name was passed and there is no "
+            "primary-aliased index for source 'dspace'."
+        ),
```

Comment:
+1 to including `match` when possible; helps to understand the exception thrown without digging into that code.